packages/playground/ssr-html/server.js

```diff
@@ -42,15 +50,30 @@ async function createServer(
   // use vite's connect instance as middleware
   app.use(vite.middlewares)
 
-  app.use('*', async (req, res) => {
+  app.use('*', async (req, res, next) => {
     try {
       let [url] = req.originalUrl.split('?')
       if (url.endsWith('/')) url += 'index.html'
 
+      if (url.startsWith('/favicon.ico')) {
+        return res.status(404).end('404')
+      }
+      if (url.startsWith('/@id/__x00__')) {
+        return next()
+      }
+
       const htmlLoc = resolve(`.${url}`)
-      let html = fs.readFileSync(htmlLoc, 'utf8')
-      html = html.replace('</body>', `${DYNAMIC_SCRIPTS}</body>`)
-      html = await vite.transformIndexHtml(url, html)
+      let template = fs.readFileSync(htmlLoc, 'utf-8')
+
+      template = template.replace(
+        '</body>',
+        `${DYNAMIC_SCRIPTS}${DYNAMIC_STYLES}</body>`
+      )
+
+      // Force calling transformIndexHtml with url === '/', to simulate
+      // usage by ecosystem that was recommended in the SSR documentation
+      // as `const url = req.originalUrl`
+      const html = await vite.transformIndexHtml('/', template)
```

Comment:
I'm forcing `'/'` as the `url` here as @brillout described he was doing in vite-plugin-ssr.